lib/mappers/esdn_mapper.py

```diff
@@ -0,0 +1,331 @@
+
+    # <note type=”regional council”>
+    def map_intermediate_provider(self):
+        prop = self.root_key + "note"
+        intermediateProvider = None
+
+        if exists(self.provider_data, prop):
+            for s in iterify(getprop(self.provider_data, prop)):
+                try:
+                    if s.get("type") == "regional council":
+                        intermediateProvider = s.get("#text")
+                except:
+                    pass
+            if intermediateProvider:
+                self.mapped_data.update({"intermediateProvider": intermediateProvider})
+
+    # Copied from DigitialNC
+    def map_is_part_of(self):
+        prop = self.root_key + "relatedItem"
+        _dict = {"isPartOf": []}
+
+        if exists(self.provider_data, prop):
+            for s in iterify(getprop(self.provider_data, prop)):
+                try:
+                    if s.get("type") == "host" and s.get("displayLevel") == "collection":
+                        _dict["isPartOf"].append(getprop(s, "titleInfo/title"))
+                except:
+                    pass
+
+            self.update_source_resource(self.clean_dict(_dict))
+
+    # <location><url usage="primary display" access="object in context">
+    # <location><url access="preview">
+    def map_is_shown_at_and_preview(self):
+        prop = self.root_key + "location"
+        _dict = {"isShownAt": "",
+                 "object": ""}
+
+        if exists(self.provider_data, prop):
+            for s in iterify(getprop(self.provider_data, prop)):
+                try:
+                    url = s.get("url")
+                    if url.get("usage") == "primary display" and url.get("access") == "object in context":
+                        _dict["isShownAt"] = url.get("#text")
+                    elif url.get("access") == "preview":
+                        _dict["object"] = url.get("#text")
+                except:
+                    pass
+
+            self.mapped_data.update(self.clean_dict(_dict))
+
+    #
+    def map_language(self):
+        prop = self.root_key + "language/languageTerm"
+        lang = None
+        if exists(self.provider_data, prop):
+            lang = getprop(self.provider_data, prop)
+
+            if isinstance(lang, dict):
+                try:
+                    lang = lang.get("#text")
+                except:
+                    pass
+
+        if lang:
+            self.update_source_resource({"language": lang})
+
+    def map_relation(self):
+        prop = self.root_key + "relatedItem/titleInfo/title"
+
+        if exists(self.provider_data, prop):
+            self.update_source_resource({"relation":
+                                         getprop(self.provider_data, prop)})
+
+    #
+    def map_rights(self):
+        prop = self.root_key + "accessCondition"
+        rights = None
+        if exists(self.provider_data, prop):
+            rights = getprop(self.provider_data, prop)
+            if isinstance(rights, dict):
```

Comment:
Under what condition would `rights` not be a dictionary? Would we want to handle that condition?